### PR TITLE
Add basic application example

### PR DESCRIPTION
**Pull request by tones111**
This is an attempt at addressing issue #79.  I'm new to rust and gtk so feedback on improvements are welcome.

The need for the new_app function is unfortunate, but all my attempts at using cfg! result in non-compiling code because the false branch of the conditional doesn't get optimized out early enough during compilation.

I probably missed something, but it's also unfortunate that the gio dependency is needed for the call to gtk::Application::new.  Is there any mechanism to access the constant through the gtk crate?

### CODE REVIEW

File: src/basic_application.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+extern crate gio;
+extern crate gtk;
+
+use gtk::prelude::*;
+
+const APP_ID: &'static str = "org.gtk-rs.basic_app";
+const APP_FLAGS: gio::ApplicationFlags = gio::APPLICATION_FLAGS_NONE;
+
+#[cfg(feature = "gtk_3_10")]
```
Comment (EPashkin) [inline src/basic_application.rs]: In https://github.com/gtk-rs/gtk/blob/master/src/application.rs option `application_id` start from v3_6.
https://developer.gnome.org/gtk3/stable/GtkApplication.html#gtk-application-new says same

**Comment by tones111** [comment]
I agree with you, but when building the examples the feature for the gtk version doesn't get passed to the compiler.  It sees the gtk_3_x features, but only gets passed a "gtk" feature.  Are you requesting I add a gtk_3_6 feature to the examples?

File: src/basic_application.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            println!("clicked");
+        });
+        window.add(&button);
+
+        window.show_all();
+    });
+
+    // use -h to see usage options
+    let args: Vec<String> = std::env::args().collect();
+    let args: Vec<&str> = args.iter().map(|x| &x[..]).collect();
```
Comment (EPashkin) [inline src/basic_application.rs]: Why not do it in single pass?

**Comment by EPashkin** [comment]
Yes.

**Comment by EPashkin** [comment]
Bigger problem: examples PRs must go against `pending` branch, not master.
Recreate it pls.

File: src/basic_application.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/basic_application.rs]: Maybe because he wanted a `Vec<&str>` instead of a `Vec<String>`?

Comment (EPashkin) [inline src/basic_application.rs]: I mean `std::env::args().map(|x| &x[..]).collect();`

Comment (GuillaumeGomez) [inline src/basic_application.rs]: In your code, `x` doesn't live long enough.

Comment (EPashkin) [inline src/basic_application.rs]: Oh, that reason for multiple lines. Forgot about that.

**Pull request closed without merging**